volplugin/utils.go

```diff
@@ -71,28 +66,26 @@ func (dc *DaemonConfig) structsVolumeName(uc *unmarshalledConfig) (storage.Mount
 
 	driver, err := backend.NewMountDriver(volConfig.Backends.Mount, dc.Global.MountPath)
 	if err != nil {
-		return nil, nil, driverOpts, errored.Errorf("loading driver").Combine(err)
+		return nil, nil, driverOpts, errors.GetDriver.Combine(err)
 	}
 
 	driverOpts, err = dc.volumeToDriverOptions(volConfig)
 	if err != nil {
-		return nil, nil, driverOpts, errored.Errorf("converting volume to internal structure").Combine(err)
+		return nil, nil, driverOpts, errors.UnmarshalRequest.Combine(err)
 	}
 
 	return driver, volConfig, driverOpts, nil
 }
 
-func httpError(w http.ResponseWriter, message string, err error) {
-	fullError := fmt.Sprintf("%s %v", message, err)
-
-	content, errc := marshalResponse(VolumeResponse{"", fullError})
+func httpError(w http.ResponseWriter, err error) {
+	content, errc := marshalResponse(VolumeResponse{"", err.Error()})
 	if errc != nil {
-		log.Warnf("Error received marshalling error response: %v, original error: %s", errc, fullError)
+		log.Warnf("Error received marshalling error response: %v, original error: %s", errc, err.Error())
 		return
 	}
 
-	log.Warnf("Returning HTTP error handling plugin negotiation: %s", fullError)
-	http.Error(w, string(content), http.StatusInternalServerError)
+	log.Warnf("Returning HTTP error handling plugin negotiation: %s", err.Error())
+	http.Error(w, string(content), http.StatusOK)
```

Comment:
Docker doesn't actually support HTTP status codes, so I corrected it, basically.